commons-vfs2/src/test/java/org/apache/commons/vfs2/FileSystemOptionsTest.java

```diff
@@ -92,4 +96,63 @@ public void testEqualsHashCodeAndCompareTo() {
         assertEquals(expected.hashCode(), actual.hashCode());
     }
 
+    @Test
+    public void testEqualsHashCodeAndCompareToWithSftpIdentityProviderMatch() {
+        for ( int mask = 0 ; mask < 8 ; mask++ ) {
+            assertSftpOptionsEqual(
+                (mask & 1) ==  1 ? new File("/tmp/test.priv") : null,
+                (mask & 2) ==  2 ? new File("/tmp/test.pub") : null,
+                (mask & 4) ==  4 ? new byte[] {1,2,3} : null
+            );
+        }
+    }
+
+    @Test
+    public void testEqualsHashCodeAndCompareToWithSftpIdentityProviderMismatch() {
+        assertSftpOptionsNotEqual(
+            new File("/tmp/test.priv"), new File("/tmp/test.pub"), new byte[] {1,2,3},
+            new File("/tmp/test1.priv"), new File("/tmp/test.pub"), new byte[] {1,2,3}
+        );
+
+        assertSftpOptionsNotEqual(
+            new File("/tmp/test.priv"), new File("/tmp/test.pub"), new byte[] {1,2,3},
```

Comment:
Refactor string constants that are reused into locals to make it obvious when the reuse is intended.